server/lib/model/action.dart

```diff
@@ -61,3 +65,151 @@ class ChangeLineItemStatusAction implements Action {
+                ..bidAmountMicros = _bidAmount);
+        break;
+      case 2:
+        // Maximize spend bid strategy.
+        action.changeLineItemBiddingStrategyParams =
+            proto.ChangeLineItemBiddingStrategyParams()
+              ..biddingStrategy = proto
+                  .ChangeLineItemBiddingStrategyParams_BiddingStrategy
+                  .MAXIMIZE_SPEND
+              ..maximizeSpendBidStrategyParams =
+                  (proto.MaximizeSpendBidStrategyParams()
+                    ..type = proto.BiddingStrategyPerformanceGoalType.valueOf(
+                        _performanceGoal));
+        break;
+      case 3:
+        // Performance goal bid strategy.
+        action.changeLineItemBiddingStrategyParams =
+            proto.ChangeLineItemBiddingStrategyParams()
+              ..biddingStrategy = proto
+                  .ChangeLineItemBiddingStrategyParams_BiddingStrategy
+                  .PERFORMANCE_GOAL
+              ..performanceGoalBidStrategyParams =
+                  (proto.PerformanceGoalBidStrategyParams()
+                    ..type = proto.BiddingStrategyPerformanceGoalType.valueOf(
+                        _performanceGoal)
+                    ..performanceGoalAmountMicros = _goalAmount);
+        break;
+      default:
+        throw UnsupportedError(
+            '$_biddingStrategy is an invalid value for a bidding strategy.');
+    }
+    return action;
+  }
+
+  /// Changes the bidding strategy of the line item using the DV360 client.
+  @override
+  Future<void> run(DisplayVideo360Client client, Target target) async {
+    final lineItemTarget = target as LineItemTarget;
+
+    String goal;
+    String strategy;
+    switch (_biddingStrategy) {
+      case 1:
+        strategy = 'FIXED';
+        break;
+      case 2:
+        strategy = 'MAXIMIZE_SPEND';
+        break;
+      case 3:
+        strategy = 'PERFORMANCE_GOAL';
+        break;
+      default:
+        throw UnsupportedError(
+            '$_biddingStrategy is an invalid value for a bidding strategy.');
+    }
+
+    if (_biddingStrategy != 1) {
+      // If the bidding strategy is maximize spend, performance goal, or
+      // unspecified.
+      switch (_performanceGoal) {
+        case 1:
+          // CPA goal.
+          goal = 'BIDDING_STRATEGY_PERFORMANCE_GOAL_TYPE_CPA';
+          break;
+        case 2:
+          // CPC goal.
+          goal = 'BIDDING_STRATEGY_PERFORMANCE_GOAL_TYPE_CPC';
+          break;
+        case 3:
+          // Viewable impressions goal.
+          if (_biddingStrategy == 2) {
+            // If maximize spend strategy:
+            goal = 'BIDDING_STRATEGY_PERFORMANCE_GOAL_TYPE_AV_VIEWED';
+          }
+          if (_biddingStrategy == 3) {
+            // If performance goal strategy
+            goal = 'BIDDING_STRATEGY_PERFORMANCE_GOAL_TYPE_VIEWABLE_CPM';
+          }
+          break;
+        default:
+          throw UnsupportedError(
```

Comment:
Maybe `$_performanceGoal is an unsopported performance goal.`? What is an example for `$_performanceGoal`?

Reply:
`_performanceGoal` is an integer value corresponding to the enum value of `BiddingStrategyPerformanceGoalType`. So 1, 2, or 3.

Reply:
Does user know what these 1, 2,3 are mapped to? Can we return the name instead of number to make the error msg more helpful?

Reply:
Yes, sure!